src/views/List.jsx

```diff
@@ -1,13 +1,50 @@
+import { useEffect, useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data, listToken }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState([]);
+
+	// Function to update the filteredData based on the searchTerm
+	const updateFilteredData = () => {
+		const filteredItems = data.filter((item) => {
+			if (item.name) {
+				return item.name.toLowerCase().includes(searchTerm.toLowerCase());
+			}
+		});
+		setFilteredData(filteredItems);
+	};
+
+	// Update filteredData whenever searchTerm or data changes
+	useEffect(() => {
+		updateFilteredData();
+	}, [searchTerm, data]);
+
+	const handleClear = () => {
+		setSearchTerm('');
+		setFilteredData(data);
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form className="filter-form">
+				<label htmlFor="search-term">Filter items:</label>
+				<input
+					type="text"
+					id="search-term"
+					value={searchTerm}
+					placeholder="Start typing here.."
+					onChange={(event) => setSearchTerm(event.target.value)}
+				></input>
```

Comment:
I'm super rusty with React, so bear with me, but I think we can simplify and just close out `input` with `/>`?